wrapper.js

```diff
@@ -71,21 +92,21 @@ function setupMethods (soljson) {
   if ('_compileJSONCallback' in soljson) {
     var compileInternal = soljson.cwrap('compileJSONCallback', 'string', ['string', 'number', 'number']);
     compileJSONCallback = function (input, optimize, readCallback) {
-      return runWithReadCallback(readCallback, compileInternal, [ input, optimize ]);
+      return runWithCallbacks(readCallback, compileInternal, [ input, optimize ]);
     };
   }
 
   var compileStandard = null;
   if ('_compileStandard' in soljson) {
     var compileStandardInternal = soljson.cwrap('compileStandard', 'string', ['string', 'number']);
     compileStandard = function (input, readCallback) {
-      return runWithReadCallback(readCallback, compileStandardInternal, [ input ]);
+      return runWithCallbacks(readCallback, compileStandardInternal, [ input ]);
```

Comment:
Hm, this will, fail, won't it?

Reply:
Actually realised the PR omitted the check for version. As we agreed two different ways are needed to pass callbacks based on the compiler version.

Reply:
True, this and the other 2 above it need the old `readWithReadCallback`

Reply:
All of them need it. For these above you can hardcode the old interface and make it version dependent under `solidity_compile`.

Reply:
Why not only these two? The others don't have the read callback.

Reply:
I meant all of the ones using `runWithCallbacks`.